plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -313,7 +316,7 @@ private static void checkLengthInCodePoints(Slice value, Type characterDataType,
 
     public static SliceWriteFunction varcharWriteFunction()
     {
-        return (statement, index, value) -> statement.setString(index, value.toStringUtf8());
+        return SliceWriteFunction.of(Types.VARCHAR, (statement, index, value) -> statement.setString(index, value.toStringUtf8()));
```

Comment:
Any reason why not NVARCHAR? Please explain.

Reply:
Not supported by Postgres (idk whether anyone is actually using this constant?)

Reply:
That may cause compatibility issues. For example, some other database like PostgreSQL may choose to support NVARCHAR but not VARCHAR. To make default work by default, we may need to revert typed nulls.

Reply:
I disagree. I didn't found any case in which this would not work. Prove me wrong on this one.

Reply:
Let's go with a VARCHAR. I am however concerned with the solution being "empirically proven".
This means we may need to retract -- or be tempted to retract -- some of these changes when we work on some new connector.

Do we have a test ensuring all oracle write functions have a null type set?

Reply:
We don't. Should I create it in BCT? Or TypeMapping or create new `TestOracleClient` maybe?